src/logger.cc

```diff
@@ -20,36 +20,29 @@ NAN_METHOD(setLevel) {
     return Nan::ThrowError(Nan::Error("Provide level"));
   }
 
-  const int64_t numberValue = Nan::To<int64_t>(info[0]).FromJust();
-  spdlog::level::level_enum level;
-  switch (numberValue) {
-    case spdlog::level::critical:
-      level = spdlog::level::critical;
-      break;
-    case spdlog::level::err:
-      level = spdlog::level::err;
-      break;
-    case spdlog::level::warn:
-      level = spdlog::level::warn;
-      break;
-    case spdlog::level::info:
-      level = spdlog::level::info;
-      break;
-    case spdlog::level::debug:
-      level = spdlog::level::debug;
-      break;
-    case spdlog::level::trace:
-      level = spdlog::level::trace;
-      break;
-    case spdlog::level::off:
-      level = spdlog::level::off;
-      break;
-    default:
-      return Nan::ThrowError(Nan::Error("Invalid level"));
+  const int64_t levelNumber = Nan::To<int64_t>(info[0]).FromJust();
+  if (levelNumber >= spdlog::level::n_levels || levelNumber < 0) {
+    return Nan::ThrowError(Nan::Error("Invalid level"));
   }
+  auto level = static_cast<spdlog::level::level_enum>(levelNumber);
+
   spdlog::set_level(level);
 }
 
+NAN_METHOD(setFlushOn) {
+  if (!info[0]->IsNumber()) {
+    return Nan::ThrowError(Nan::Error("Provide flush level"));
+  }
+
+  const int64_t levelNumber = Nan::To<int64_t>(info[0]).FromJust();
+  if (levelNumber >= spdlog::level::n_levels || levelNumber < 0) {
```

Comment:
```suggestion
  if (levelNumber >= spdlog::level::n_levels || levelNumber < spdlog::level::trace) {
```

this allows to catch any change in enum constants.